File: topics/control-flow/exercises/grid.c
void draw_grid_pattern(int num_rows, int num_cols)
{
    int x, y, i, j, bar_width, bar_height;
    
    i = 0;
    bar_width = screen_width() / num_cols;
    bar_height = screen_height() / num_rows;
    
    while ( i < num_rows )
    {
        y = i * bar_height;
        j = 0;
        
        while ( j < num_cols )
        {
            x = j * bar_width;
            if ( ((i + j) % 2) == 0 )
                fill_rectangle(ColorWhite, x, y, bar_width, bar_height);
            else
                fill_rectangle(ColorBlack, x, y, bar_width, bar_height);
            
            j++;
        }
        i++;
    }
    refresh_screen();
}

File: topics/control-flow/exercises/Grid.pas
procedure DrawGridPattern(numRows, numCols: Integer);
var
    x, y, i, j, barWidth, barHeight: Integer;
begin
    i = 0;
    barWidth = screen_width() / numCols;
    barHeight = screen_height() / numRows;
    
    while i < numRows do
    begin
        y := i * barHeight;
        j := 0;
        
        while j < numCols do
        begin
            x := j * barWidth;
            if  ((i + j) mod 2) = 0 then
                fill_rectangle(ColorWhite, x, y, barWidth, barHeight)
            else
                fill_rectangle(ColorBlack, x, y, barWidth, barHeight);
            
            j += 1;
        end;
        i += 1;
    end;
    RefreshScreen();
end;